service/cziti/dns-proxy.go

```diff
@@ -260,24 +260,21 @@ func cleanDomainsForNrpt() map[string]bool {
 	return domainMap
 }
 
-func runDNSproxy(upstreamDnsServers []string, localDnsServers []net.IP) {
-	windns.FlushDNS() //do this in case the services come back in different order and the ip returned is no longer the same
-	log.Infof("starting DNS proxy upstream: %v, local: %v", upstreamDnsServers, localDnsServers)
-	domains = windns.GetConnectionSpecificDomains()
-	log.Infof("ConnectionSpecificDomains detected: %v", domains)
+func runDNSproxy(localDnsServers []net.IP) {
 	defer func() {
 		if err := recover(); err != nil {
 			log.Errorf("Recovering from panic due to DNS-related issue. %v", err)
 			dnsPanicRecover(localDnsServers, time.Now())
 		}
 	}()
-
-	domainMap := cleanDomainsForNrpt()
-	windns.AddNrptRules(domainMap, dnsip.String())
-	log.Infof("Added connection specific domains to NRPT: %v", domainMap)
-
+	windns.FlushDNS() //do this in case the services come back in different order and the ip returned is no longer the same
 	dnsRetryInterval := 500
 
+GetUpstream:
+	upstreamDnsServers := windns.GetUpstreamDNS()
+	log.Infof("starting DNS proxy upstream: %v, local: %v", upstreamDnsServers, localDnsServers)
+	AddDomainSpecificNrpt()
```

Comment:
assuming the block at line 324 can be removed, i'd argue to just put this function call back to being inline code. it's easier to read inline. if not - this should be renamed to AddConnectionSpecificDomainsToNrpt but I think just keeping the code inline is cleaner?